src/components/home.js

```diff
@@ -0,0 +1,116 @@
+import '../style/home.css';
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import searchIcon from '../assets/icons/search-icon.png';
+import cloudImg from '../assets/images/weatherIcon.png';
+import nextArrow from '../assets/icons/next.png';
+import DisplayWeatheronhome from './displayWeatheronhome';
+import { getLatLon } from '../Redux/home/homeSlice';
+import { getweatherData } from '../Redux/detail/detailSlice';
+
+const Home = () => {
+  const [country, setCountry] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const home = useSelector((state) => state.home);
+  const dispatch = useDispatch();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (country === '') {
+      setErrorMessage('Please type city name');
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+      return;
+    }
+    dispatch(getLatLon(country));
+    setCountry('');
+  };
+
+  return (
+    <>
+      <div className="form-container">
+        <form onSubmit={handleSubmit}>
+          <input
+            className="searchcity"
+            value={country}
+            onChange={(e) => setCountry(e.target.value)}
+            placeholder="Search your city"
```

Comment:
- [ ] When I try to search a City, there's no filtering that occurs. For example, I wanted to search London, but when I type in the search box, all the cities are still displayed. See screenshot below:
![image](https://github.com/hasnatali1947/Weather-App/assets/62849872/b17090c2-f84a-47e7-95e4-c6c38cbd56b5)
Kindly fix your filtering to provide just the data being searched.

Reply:
It works by searching the name
![image](https://github.com/hasnatali1947/Weather-App/assets/119807955/a13e1a40-a162-4632-b8b7-943e6844dd32)

**Hit Enter or click on the search icon**

The city would be added
![image](https://github.com/hasnatali1947/Weather-App/assets/119807955/120ea44c-e08d-4f4b-8536-e235fc15ccd6)


Also, details are working properly
![image](https://github.com/hasnatali1947/Weather-App/assets/119807955/09af2506-550b-4eb1-b879-f978aa71a223)